tests/testthat/test-select_best.R

```diff
@@ -126,6 +126,57 @@ test_that("show_best()", {
   })
 })
 
+test_that("show_best respects `eval_time` argument (#704)", {
+  skip_if_not_installed("censored")
+  skip_if_not_installed("modeldata")
+
+  library(censored)
+  library(modeldata)
+  library(rsample)
+  library(yardstick)
+  library(tibble)
+
+  data("mlc_churn")
+
+  mlc_churn <-
+    mlc_churn %>%
+    dplyr::mutate(
+      churned = ifelse(churn == "yes", 1, 0),
+      event_time = Surv(account_length, churned)
+    ) %>%
+    dplyr::select(event_time, account_length, voice_mail_plan) %>%
+    dplyr::slice(1:500)
+
+  set.seed(6941)
+  churn_rs <- bootstraps(mlc_churn, times = 2)
+
+  sr_tune_spec <- survival_reg(dist = tune())
+  eval_times <- c(10, 100, 150)
+  event_metrics <- metric_set(brier_survival, brier_survival_integrated,
+                              concordance_survival, roc_auc_survival)
+  sr_grid <- tibble(dist = c("loglogistic", "lognormal"))
+
+  sr_tune_res <-
+    sr_tune_spec %>%
+    tune_grid(
+      event_time ~ .,
+      resamples = churn_rs,
+      metrics = event_metrics,
+      eval_time = eval_times,
+      grid = sr_grid
+    )
+
+
+  expect_s3_class_bare_tibble(
+    show_best(sr_tune_res, metric = "brier_survival_integrated", eval_time = 10)
+  )
```

Comment:
This test is pretty minimal for now because I'm not sure the function _does_ respect `eval_time`:

``` r
library(tidymodels)
library(censored)
#> Loading required package: survival

tidymodels_prefer()
theme_set(theme_bw())
options(pillar.advice = FALSE, pillar.min_title_chars = Inf)

data("mlc_churn")

mlc_churn <-
  mlc_churn %>%
  mutate(
    churned = ifelse(churn == "yes", 1, 0),
    event_time = Surv(account_length, churned)
  ) %>%
  select(event_time, account_length, voice_mail_plan) %>%
  slice(1:500)

set.seed(6941)
churn_split <- initial_split(mlc_churn, prop = 4/5)
churn_tr <- training(churn_split)
churn_te <- testing(churn_split)
churn_rs <- bootstraps(churn_tr, times = 2)

sr_tune_spec <- survival_reg(dist = tune())
eval_times <- c(10, 100, 150)
event_metrics <- metric_set(brier_survival, brier_survival_integrated,
                            concordance_survival, roc_auc_survival)
sr_grid <- tibble(dist = c("loglogistic", "lognormal"))

sr_tune_res <-
  sr_tune_spec %>%
  tune_grid(
    event_time ~ .,
    resamples = churn_rs,
    metrics = event_metrics,
    eval_time = eval_times,
    grid = sr_grid,
    control = control_grid(save_pred = TRUE)
  )

show_best(sr_tune_res, metric = "brier_survival_integrated", eval_time = 10)
#> # A tibble: 2 × 8
#>   dist        .metric         .estimator .eval_time   mean     n std_err .config
#>   <chr>       <chr>           <chr>           <dbl>  <dbl> <int>   <dbl> <chr>  
#> 1 loglogistic brier_survival… standard           NA 0.0282     2 0.00591 Prepro…
#> 2 lognormal   brier_survival… standard           NA 0.0399     2 0.00479 Prepro…
show_best(sr_tune_res, metric = "brier_survival_integrated", eval_time = 100)
#> # A tibble: 2 × 8
#>   dist        .metric         .estimator .eval_time   mean     n std_err .config
#>   <chr>       <chr>           <chr>           <dbl>  <dbl> <int>   <dbl> <chr>  
#> 1 loglogistic brier_survival… standard           NA 0.0282     2 0.00591 Prepro…
#> 2 lognormal   brier_survival… standard           NA 0.0399     2 0.00479 Prepro…
show_best(sr_tune_res, metric = "brier_survival_integrated", eval_time = 15)
#> # A tibble: 2 × 8
#>   dist        .metric         .estimator .eval_time   mean     n std_err .config
#>   <chr>       <chr>           <chr>           <dbl>  <dbl> <int>   <dbl> <chr>  
#> 1 loglogistic brier_survival… standard           NA 0.0282     2 0.00591 Prepro…
#> 2 lognormal   brier_survival… standard           NA 0.0399     2 0.00479 Prepro…
```

<sup>Created on 2023-10-06 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

Is that right? If so, let's open another issue.